runelite-api/src/main/java/net/runelite/api/Player.java

```diff
@@ -53,4 +59,88 @@ public PlayerComposition getPlayerComposition()
 	{
 		return new PlayerComposition(player.getComposition());
 	}
+
+	public Model getModel()
+	{
+		return player.getModel();
+	}
+
+	public Polygon[] getPolygons()
+	{
+		ArrayList<Polygon> polys = new ArrayList<>();
+		Model model = getModel();
+		int localX = player.getX();
+		int localY = player.getY();
+		int orientation = (player.getOrientation() + 1024) % 2048;
+
+		if (model == null)
+		{
+			return null;
+		}
+
+		int[] verticesX = model.getVerticesX();
```

Comment:
Copy these here, and remove originalX/Y, that way the underlying model isn't modified.